### ISSUE TITLE
Automatic creation of GitHub OAuth token (via browser) from CLI

### ISSUE THREAD

**Issue by theefer** [jspm/jspm-cli#848]
The user flow to setup GitHub auth (e.g. following rate limiting) is slightly cumbersome, especially when trying to use an access token (safer, and the only option if 2FA is enabled). You have to open a browser, guess what scopes are needed, copy the token, etc.

Is there any argument against auto-creating a token when running `jspm registry config github` by opening a web browser, getting the user to log into GitHub and automatically retrieving the newly generated token? For instance it could use a library to do so such as [github-oauth](https://www.npmjs.com/package/github-oauth).

I have watched several people struggle with this in the last few days and it would be great to reduce the friction around this.

**Comment by guybedford** [comment]
This would be great!

**Comment by OrKoN** [comment]
@guybedford it requires an app on Github and the app secret and id must be included in JSPM. Is it secure?

**Comment by theefer** [comment]
@OrKoN You're right, that's also what I concluded after trying it so I don't think that option is viable.

I think the only solution would be to prompt the user for their username, password and optionally 2FA on the CLI and use those to create an OAuth token with the necessary permissions. The password would therefore not be saved, only the token. I've got a small proof-of-concept working but haven't integrated with JSPM yet. I also saw that 0.16 has something similar integrated so I need to check what it does in the latest beta and see what, if anything, can still be improved upon.
